docs/mathematics/armstrong-number.mdx

```diff
@@ -0,0 +1,170 @@
+
+Result : 1634 is an Armstrong Number.
+
+## Algorithm
+
+ - Input the number `num`.
+ - Calculate the number of digits in `num`. Let this be stored in `order`.
+ - For each digit `d` in `num`, compute `d` raised to the power of `order`.
+ - Calculate the sum of each `d` to the power of `order` for all the digits in `num` and store in variable `sum`.
+ - Compare `sum` and `num`. If equal, return `An Armstrong Number`. Else, return `Not an Armstrong Number`.
+
+## Code 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+// Function to check whether the Number is Armstrong Number or Not.
+bool is_armstrong(int n)
+{
+	if (n < 0)
+	{
+		return false;
+	}
+	int sum = 0;
+	int var = n;
+	int number_of_digits = floor(log10(n) + 1);
+	while (var > 0)
+	{
+		int rem = var % 10;
+		sum = sum + pow(rem, number_of_digits);
+		var = var / 10;
+	}
+	return n == sum;
+}
+int main()
+{
+	cout << "Enter the Number: " << endl;
+	int n;
+	cin >> n;
+	if (is_armstrong(n))
+		cout << n << " is an Armstrong Number." << endl;
+	else
+		cout << n << " is not an Armstrong Number." << endl;
+	return 0;
+}
+```
+</TabItem>
+
+<TabItem value="py">
+
+```python
+# This program checks whether a number is an armstrong number or not
+import math
+def no_of_digits(n : int) -> int:
+	count = 0
+	while n > 0:
+		count += 1
+		n = n // 10
+	return count
+
+def add(n : int) -> int:
+	temp = n
+	tot = 0
+	nod = no_of_digits(temp)
+	while n > 0:
+		r = n % 10
+		tot += r ** nod
+		n = n // 10
+	return tot
+
+def checkArmstrong(num : int) -> bool:
+	return add(num) == num
+
+num = int(input("Enter the number: "))
+if checkArmstrong(num):
+	print(num," is an Armstrong Number.")
+else:
+	print(num," is not an Armstrong Number.")
+```
+</TabItem>
+
+<TabItem value="java">
+
+```java
+import java.util.*;
+public class ArmstrongNumber
+{
+	static int length_number(int num){
+		return String.valueOf(num).length();
+	}
+
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		System.out.print("Enter the number: ");
+		int num = sc.nextInt();
+		int res = num;
+		int val = 0;
+		int len = length_number(num);
+		while(res>0){
+			int temp = res % 10;
+			val += Math.pow(temp, len);
+			res = res / 10;
+		}
+		if(num == val) System.out.println(num+" is an Armstrong Number.");
+		else System.out.println(num+" is not an Armstrong Number.");
+	}
+
+}
```

Comment:
A small refactor: Can you please shift the actual logic to some other function rather than the `main()` function. This should not have ideally happened but since a lot of code on the NeoAlgo was merged without any review whatsoever, these errors crept over.

You can make a PR to add the refactored code so that you can be compensated for your efforts.

Reply:
Hey, @HarshCasper I have done all the changes and have added the PR for the refactored code here : 
https://github.com/TesseractCoding/NeoAlgo/pull/5349
Please review them. Thank you :)